### ISSUE TITLE
gpr_thd_test failure: assertion failed: gpr_thd_new(&thds[i], &thd_body_joinable, NULL, &options)

### ISSUE THREAD

**Issue by jcanizales** [grpc/grpc#5413]
See https://grpc-testing.appspot.com/job/gRPC_portability_master/442/language=c,scenario=linux_x86_default/testReport/(root)/tests/bins_dbg_gpr_thd_test/

```
thd_test.c:101]             assertion failed: gpr_thd_new(&thds[i], &thd_body_joinable, NULL, &options)
*******************************
Caught signal 6
```

**Comment by murgatroid99** [comment]
Almost identical failure at https://grpc-testing.appspot.com/job/gRPC_portability_master/480/language=c,scenario=linux_x86_default/testReport/(root)/tests/bins_dbg_gpr_thd_test/

```
E0226 21:04:34.098648533   11812 thd_test.c:91]              assertion failed: gpr_thd_new(&thd, &thd_body, &t, NULL)
*******************************
Caught signal 6
```

**Comment by nathanielmanistaatgoogle** [comment]
[Seen again today](https://grpc-testing.appspot.com/job/gRPC_portability_master/762/language=c,scenario=linux_x86_default/testReport/(root)/tests/bins_dbg_gpr_thd_test/).

**Comment by jtattermusch** [comment]
@sreecha , @dgquintas  this is one out of very few flakes that preventing our portability suite from becoming green. Any chance you could take a look. I think this one shouldn't be hard to fix.

Please note it's x86 specific.

**Comment by sreecha** [comment]
Looks like p_thread_create() function is hitting a memory resource limit - (primarily for each thread stack)

When I debugged this a bit on my dev machine, I saw that it was able to create around 509 threads before it fails (and that number is consistent across repeated runs on my machine). I could make this test pass by setting the thread stack size to a lower number (512K)..

Let me see if there is a docker option to give more memory or set the default stack size to a lower number (just for this test). If not, lowering the # of threads seems like a reasonable solution.
